I have added the ability to register and log in users to our REST API,

Travellin91 opened the pull request:
I have added the ability to register and log in users to our REST API, using JWT for security. Now users can create an account, log in, obtain an access token, and check their status within the system. Additionally, we have ensured access protection to resources where necessary, by verifying tokens and au

Travellin91 committed:
I have added the ability to register and log in users to our REST API,

I have added the ability to register and log in users to our REST API, using JWT for security. Now users can create an account, log in, obtain an access token, and check their status within the system. Additionally, we have ensured access protection to resources where necessary, by verifying tokens and au

ceREAL133 commented on app.js:
```diff
@@ -1,25 +1,45 @@
+app.use(cors());
+app.use(express.json());
+
+const uriDb = `mongodb+srv://${encodeURIComponent(process.env.DB_USERNAME)}:${encodeURIComponent(process.env.DB_PASSWORD)}@${process.env.DB_HOST}/db-contacts?retryWrites=true&w=majority`;
```
this is sensitive info. hide it in .env

ceREAL133 replied:
all the mongoDBUri must be inside .env file

ceREAL133 commented on routes/api/authRouter.js:
```diff
@@ -0,0 +1,96 @@
+
+router.post("/users/register", async (req, res) => {
+  const { error } = registerSchema.validate(req.body);
+  if (error) {
```
functions inside routes must be separated in controller files. And just be invoked here

ceREAL133 commented on routes/api/contacts.js:
```diff
@@ -1,25 +1,77 @@
+router.get("/:id", authMiddleware, async (req, res) => {
+  const contactId = req.params.id;
+  const userId = req.user.id;
+  try {
```
same here. You have contacts controller file. But still got all functions inside routes. What's the point of having controller then?

ceREAL133 replied:
your routes should look like:
`router.get("/:id", authMiddleware, getUserById)`

and that's all

Travellin91 committed:
fixed

Separated the functions responsible for handling contact requests into a separate controller file named contactsController.

Utilized controllers in the routes:
Replaced the functions within the routes with calls to the corresponding functions from the contactsController.

Applied authentication checks to all previously configured routes:
Added the authMiddleware to all routes that require authentication to ensure the security of user data.
